### PR TITLE
test(spanner)!: fix data race in spanner integration tests

### PR DESCRIPTION
**Pull request by rahul2393**
* Update GFELatencyMetricsEnabled flag private to the package, to avoid possibility of data race from customers.
* Added setter/getter method to update the GFELatencyMetricsEnabled flag.
* Added lock to make GFELatencyMetricsEnabled thread safe.

Fixes https://github.com/googleapis/google-cloud-go/issues/5272

### CODE REVIEW

File: spanner/stats.go [hunk 1]
```diff
@@ -213,28 +216,40 @@ func EnableStatViews() error {
...
 func EnableGfeLatencyAndHeaderMissingCountViews() error {
-	GFELatencyMetricsEnabled = true
+	SetGFELatencyMetricsFlag(true)
 	return view.Register(
 		GFELatencyView,
 		GFEHeaderMissingCountView,
 	)
 }
 
+func GetGFELatencyMetricsFlag() bool {
```
Comment (olavloite) [inline spanner/stats.go:238]: Could this be unexported? And same for the setter?

**Review by asthamohta — APPROVED** [review]

File: spanner/stats.go [hunk 2]
```diff
@@ -267,12 +282,6 @@ func checkCommonTagsGFELatency(t *testing.T, m map[tag.Key]string) {
 	if !strings.HasPrefix(m[tagKeyClientID], "client") {
 		t.Fatalf("Incorrect client ID: %v", m[tagKeyClientID])
 	}
-	if !strings.HasPrefix(m[tagKeyInstance], "gotest") {
```
Comment (olavloite) [inline spanner/stats.go:270]: Good catch 👍

File: spanner/stats.go (hunk 1 above)
Comment (rahul2393) [reply spanner/stats.go]: done

**Comment by olavloite** [comment]
By the way: The build error seems to be unrelated to this change. AFAICT, it is caused by a formatting error in the unedited `iam/go_mod_tidy_hack.go` file.

@codyoss It seems that #5269 introduced a formatting error that was hidden by a different build error in that PR, but that it is a formatting error that only seems to be picked up by Go 1.17. At least, if I run `gofmt -s -d` on that file using Go 1.16, it thinks it's ok.

**Review by olavloite — APPROVED** [review]
If I understand it correctly, the GFE Latency feature had not yet been released. We should make sure that this change then is included in the next release, as it would otherwise be a breaking change (we are removing an exported variable).

**Comment by rahul2393** [comment]
No its not yet released so its safe for now, here is the release PR https://github.com/googleapis/google-cloud-go/pull/5219

**Comment by rahul2393** [comment]
@olavloite build passed after changing iam/go_mod_tidy_hack.go and adding the unused exported GFELatencyMetricsEnabled field back to this PR. is it good to merge now?
cc: @codyoss @hengfengli

**Review by hengfengli — APPROVED** [review]
LGTM.

**Pull request merged**